protocol/xmrtaker/message_handler.go

```diff
@@ -239,68 +236,39 @@ func (s *swapState) handleNotifyXMRLock(msg *message.NotifyXMRLock) (net.Message
 
 	t := time.Now().Format(common.TimeFmtNSecs)
 	walletName := fmt.Sprintf("xmrtaker-viewonly-wallet-%s", t)
-	if err := s.GenerateViewOnlyWalletFromKeys(vk, kp.Address(s.Env()), walletName, ""); err != nil {
+	if err := s.GenerateViewOnlyWalletFromKeys(vk, lockedAddr, s.moneroBlockHeight, walletName, ""); err != nil {
 		return nil, fmt.Errorf("failed to generate view-only wallet to verify locked XMR: %w", err)
 	}
 
 	log.Debugf("generated view-only wallet to check funds: %s", walletName)
 
-	if s.Env() != common.Development {
-		log.Infof("waiting for new blocks...")
-		// wait for 2 new blocks, otherwise balance might be 0
-		// TODO: check transaction hash (#164)
-		height, err := monero.WaitForBlocks(s.Backend, 2)
-		if err != nil {
-			return nil, err
-		}
-
-		log.Infof("monero block height: %d", height)
-	}
-
-	log.Debug("refreshing client...")
-
 	if err := s.Refresh(); err != nil {
 		return nil, fmt.Errorf("failed to refresh client: %w", err)
 	}
 
-	accounts, err := s.GetAccounts()
```

Comment:
We created the A+B swap wallet and know the funds are in the first and only account index, so I simplified this code. I tried to find a way to use `generate_from_keys` to create a non-zero index in an existing wallet, but it appears to not be possible.